app/src/main/java/com/mr/nemo/dragonfly/ui/screen/auth/signup/SignUpFieldPage.kt

```diff
@@ -0,0 +1,222 @@
+package com.mr.nemo.dragonfly.ui.screen.auth.signup
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.mr.nemo.dragonfly.R
+import com.mr.nemo.dragonfly.domain.entity.auth.signup.PasswordValue
+import com.mr.nemo.dragonfly.domain.entity.auth.signup.PhoneValue
+import com.mr.nemo.dragonfly.domain.entity.auth.signup.SignUpField
+import com.mr.nemo.dragonfly.ui.component.text.TitleText
+import com.mr.nemo.dragonfly.ui.component.textfield.BaseTextField
+import com.mr.nemo.dragonfly.ui.component.textfield.drowdown.LargeDropdownMenu
+import com.mr.nemo.dragonfly.ui.theme.DragonFlyTheme
+
+@Composable
+fun <T, F : SignUpField<T>> SignUpFieldPage(
```

Comment:
@bohdanzdaniekin isn't it kinda overengeneering? 🤔

Reply:
@slothdeboss Probably yes, but for a while, I didn't come up with any other solution and I'm probably going to refactor it later